### DISCUSSION TITLE
Endpoints/Paths used by Bitwarden Browser Extension

### DISCUSSION THREAD

**Discussion by atefhaloui** [dani-garcia/vaultwarden#3188] [Q&A]
Hi,
    I would like to expose a new service on my network but with some restrictions:
* The web portal is only available through a 2FA (using authelia)
* The access with bitwarden browser extension shoudl work without 2FA.
* the /admin path is not available.
I'm trying the following nginx configuration but the browser extension is not working. What are the list paths/endpoints required so that the extension works correctly please ?

```
server {
  listen 443 ssl;
  server_name xxxxx;

  ssl_certificate     /etc/ssl/private/xxxxx.crt;
  ssl_certificate_key /etc/ssl/private/xxxxx.key;

  access_log /var/log/nginx/xxxxx_access.log;
  error_log  /var/log/nginx/xxxxx_error.log;

  # Admin is only available through intranet
  location = /admin {
    return 301 /;
  }

  location ~ ^/(api|location|notifications|icons)/(.*)$ {
    include /etc/nginx/conf.d/proxy.conf;

    proxy_pass http://upstream_vaulwarden/$1/$2;
  }

  location / {
    include /etc/nginx/conf.d/proxy.conf;
    include /etc/nginx/conf.d/authelia.conf;

    proxy_pass http://upstream_vaulwarden/;
  }
}
```

Cheers

**Comment by sorcix** [comment]
Protecting only the web interface with Authelia makes no sense. The webapp is just a bunch of static html/css/javascript using the same API as the browser extensions.

**Reply by atefhaloui** [reply]
What are the best practices to protect vaultwarden please ?
I've read that there are no option to force 2FA like TOTP for all users.

Thank you
